src/configure/mod.rs

```diff
@@ -88,7 +85,7 @@ pub unsafe fn dc_job_do_DC_JOB_CONFIGURE_IMAP(context: &Context) {
 
             const STEP_3_INDEX: u8 = 13;
             let mut step_counter: u8 = 0;
-            while !s.shall_stop_ongoing {
+            while !context.shall_stop_ongoing() {
```

Comment:
this here might actually fix a problem.  i'd think that because the lock is obtained further up, and not anymore in the looping that cancelling configure might not work nicely/really before.